### PR TITLE
:sparkles:  Java provider as external provider

### LINKED ISSUE #469: Implement external provider for Java

### PR DESCRIPTION
**Pull request by eemcmullan**
Addon PR: 82 
Closes #469

### CODE REVIEW

File: external-providers/java-external-provider/pkg/java_external_provider/provider.go [hunk 1]
```diff
@@ -332,43 +314,43 @@ func (p *javaProvider) Init(ctx context.Context, log logr.Logger, config provide
...
-			},
-		}, nil
-	}
-
-	defer func() {
-		j.depsMutex.Lock()
-		j.depsLocationCache[cacheKey] = location.StartPosition.Line
-		j.depsMutex.Unlock()
-	}()
+	//cacheKey := fmt.Sprintf("%s-%s-%s-%v",
```
[the remark this answers is not in the document]
Comment (pranavgaikwad) [reply external-providers/java-external-provider/pkg/java_external_provider/provider.go:329]: We need this! this is to get locations from pom.xml for a dependency condition.

[the remark this answers is not in the document]
Comment (eemcmullan) [reply external-providers/java-external-provider/pkg/java_external_provider/provider.go:329]: Sorry, leaving this commented out was not intentional - did so to debug and forgot to revert it.

**Comment by eemcmullan** [comment]
@shawn-hurley Dep analysis seems to be working

**Comment by shawn-hurley** [comment]
I assume that the removal of the dep code snip is failing the e2e tests. I think that we may not be able to make this change as a follow up.

**Pull request closed without merging**